src/react/utils.js

```diff
@@ -1003,7 +1035,9 @@ export function applyObjectAssignConfigsForReactElement(realm: Realm, to: Object
                 Properties.Set(realm, to, propName, Get(realm, source, propName), true);
               }
             }
-            delayedSources.push(source.getSnapshot());
+            let snapshot = source.getSnapshot();
+            delayedSources.push(snapshot);
+            source.temporalAlias = snapshot;
```

Comment:
This looks suspicious to me. We're mutating the source here? What if the same source is used in multiple calls, or even many times in a single call? We'd lose this information for all but the last call. So it seems inconsistent.

Reply:
This is actually another correctness thing, that wasn't done before but should have been. See the original snapshot logic from Object.assign:

https://github.com/facebook/prepack/blob/master/src/intrinsics/ecma262/Object.js#L122

Reply:
Do we know it improves correctness or is it a guess? It could have been a bug in the original code too. Can you explain in more detail what specifically does doing this solve?

Reply:
Sure, happy to explain as it took me a while to understand it too. What I meant by a correctness change, was that I wanted the React logic to 1:1 match the existing logic of Object.assign. When we snapshot an object via `getSnapshot` we first check if the object already has a `temporalAlias` value, if so we return that value otherwise we do some work and create a temporal of that object at that point in time. This is important, because `_temporalAlias` is a tracked value on the object, which means that it's bindings can change between different scopes/effects.

So at a later point, if we try to call `getSnapshot` again and the object has already had it's temporalAlias set, then we use that value. That value may have been created in some effects with a conditional, so the value might be a conditional abstract representing two possible temporal values at different points. If we overwrote the temporal alias and removed the existing conditional, we'd potentially break things.